// knowledge-base/webpack.config.js
const path = require('path')
const VueLoaderPlugin = require('vue-loader/lib/plugin')

module.exports = {
    devServer: {
        contentBase: path.join(__dirname, 'dist'),
        compress: true,
        port: 9000
      },
    entry: "./src/main.js",
    mode:"development",
    output:{
        path: path.resolve("./dist/"),
        filename: "bundle.js",
        publicPath:"dist/"
    },
    plugins: [
        new VueLoaderPlugin()
    ],
    module:{
        rules:[
            {
                test: /\.vue$/,
                loader:"vue-loader"
            },
            {
                test: /\.(woff2?|eot|ttf|svg)$/,
                loader: 'url-loader',
                options: {
                  limit: 1 * 1024 ** 1,
                  outputPath: "fonts/",
                },
              },
              {
                test: /\.(png|jpe?g|gif)$/,
                loader: 'url-loader',
                options: {
                  limit: 1 * 1024 ** 1,
                  outputPath: "images/",
                }
              },
              {
                test: /\.css$/,
                use: [
                  { loader: "style-loader" },
                  { loader: "css-loader" },
                ],
              },
              {
                test: /\.less$/,
                use: [
                  { loader: "style-loader" },
                  { loader: "css-loader" },
                  { loader: "less-loader" },
                ],
              },
              {
                test: /\.styl(us)?$/,
                use: [
                  'vue-style-loader',
                  'css-loader',
                  'stylus-loader'
                ]
              }
        ]
    }
    }
